src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
Agreed. I'd still like to see methods analogous `takebuf_array` and `takebuf_str`instead of having users directly reach into the `data` field, but it seems in any possible future API design, we're going to want the actual `data` to be a byte array.